src/internal_rep.rs

```diff
@@ -1702,6 +1704,22 @@ impl ValidatedCall {
     }
 }
 
+// If the class_name is "this", return the parent type name,
+// else return the class name.  If the class_name is "this", and the parent type is None, that is
+// an internal error
```

Comment:
Would it be possible to name/declare a resource "this"? What forbids it?

Reply:
Good point.  We should probably add a list of reserved keywords and ban declarations with that name.  I'll add that to the backlog.  In my opinion we shouldn't let that block this, since that's broken everywhere for all keywords.